compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCodeGenUtil.java

```diff
@@ -131,20 +132,24 @@ static void createFunctionPointer(MethodVisitor mv, String className, String lam
     }
 
     /**
-     * Cleanup type name by replacing '$' with '_'.
+     * Cleanup type name by replacing '/ ' with '_' for readonly types.
      *
      * @param name name to be replaced and cleaned
      * @return cleaned name
      */
-    static String cleanupTypeName(String name) {
```

Comment:
Analyse the reason of replacing (readonly  type -at master)

Reply:
We can't move this logic to jvm desugar phase, because some type names are called from instruction level. (Eg. `NEW_TYPEDESC`)